src/main/java/dukchess/dao/TaskList.java

```diff
@@ -0,0 +1,78 @@
+package dukchess.dao;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import dukchess.entity.Deadline;
+import dukchess.entity.Event;
+import dukchess.entity.Task;
+import dukchess.entity.Todo;
+
+public class TaskList extends ArrayList<Task> {
+    private File taskFile;
+    private FileWriter taskFileWriter;
+    private Scanner taskFileScanner;
+
+    public TaskList(File taskFile) throws IOException {
+        super();
+        this.taskFile = taskFile;
+        this.taskFileScanner = new Scanner(taskFile);
+        initializeFromFile();
+        this.taskFileWriter = new FileWriter(taskFile);
+    }
+
+    public void flushTaskList() {
+        try {
+            for (int i = 0; i < this.size(); i++) {
+                Task currentTask = this.get(i);
+                String currentLine = currentTask.toString();
+                taskFileWriter.append(currentLine).append(System.getProperty("line.separator"));
+            }
+            this.taskFileWriter.close();
+            this.taskFileScanner.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void initializeFromFile() {
+        if (taskFile == null) {
+            return;
+        }
+        Pattern taskStringPattern = Pattern.compile("\\[(\\w)\\]\\[(.?)\\] ([^(]+)(\\((.+)\\))?");
+        while (taskFileScanner.hasNext()) {
+            Task task = null;
+
+            String currentTaskString = taskFileScanner.nextLine();
+
+            Matcher taskStringMatches = taskStringPattern.matcher(currentTaskString);
+            taskStringMatches.matches();
+            String taskType = taskStringMatches.group(1);
+            boolean isDone = taskStringMatches.group(2).equals("X");
+            String taskDescription = taskStringMatches.group(3).trim();
+            String taskDetails = taskStringMatches.group(5);
+
+            switch (taskType) {
+            case "T":
```

Comment:
Using a single alphabet letter to describe a task type is less reader friendly. You could try naming improvements suggested [here](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#use-name-to-explain).